hooks/stages/entrypoint_test.go

```diff
@@ -0,0 +1,156 @@
+package stages
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/prebid/prebid-server/hooks/invocation"
+)
+
+func TestExecuteEntrypointHook_CannotModifyPayloadInsideHookCallMethod(t *testing.T) {
+	entrypointHooks = []EntrypointHook{mockNopEntrypointHook{}, mockNopEntrypointHook{}}
+
+	ctx := context.TODO()
+	body := []byte(`{"name": "John", "last_name": "Doe"}`)
+	reader := bytes.NewReader(body)
+	req, err := http.NewRequest(http.MethodPost, "https://prebid.com/openrtb2/auction", reader)
+	if err != nil {
+		t.Fatalf("Unexpected error creating http request: %s", err)
+	}
+
+	newBody, err := ExecuteEntrypointHook(ctx, invocation.Context{Endpoint: "auction", Timeout: time.Second}, req, body)
+	if err != nil {
+		t.Fatalf("Unexpected error executing entrypoint hook: %s", err)
+	}
+
+	if bytes.Compare(body, newBody) != 0 {
+		t.Errorf("Body changed inside hook.Call method, expected: %s, got: %s", string(body), string(newBody))
+	}
+}
+
+type mockNopEntrypointHook struct{}
+
+func (e mockNopEntrypointHook) Code() string {
+	return "MockedNopEntrypointHook"
+}
+
+func (e mockNopEntrypointHook) Call(ctx context.Context, invocationContext invocation.Context, payload ImmutableEntrypointPayload) (invocation.HookResult[MutableEntrypointPayload], error) {
+	_ = jsonparser.Delete(payload.Body, "name")
+	_, _ = jsonparser.Set(payload.Body, []byte(`"bar"`), "foo")
+
+	return invocation.HookResult[MutableEntrypointPayload]{Action: invocation.Nop}, nil
+}
+
+func TestExecuteEntrypointHook_CanApplyHookMutations(t *testing.T) {
+	entrypointHooks = []EntrypointHook{mockUpdateEntrypointHook{}, mockUpdateEntrypointHook{}}
+
+	ctx := context.TODO()
```

Comment:
Nitpick: For tests we should use `context.Background()` instead of `context.TODO()`. TODO is not intended for use in tests.